### ISSUE TITLE
IgnoreGroup not respected for AUR packages

### ISSUE THREAD

**Issue by mrak** [aurapm/aura#312]
In my `pacman.conf` I have set `IgnoreGroup = modified` for packages that I have overridden the default build for.

This works for the official repositories (hands off to `pacman`) but does not seem to have any effect for AUR packages.

**Comment by flamusdiu** [comment]
This is (sadly) the way it works. pacman.conf works only for official repos (one setup in pacman.conf) not AUR. =( Not sure if @fosskers  has a solution in mind for that.

**Comment by fosskers** [comment]
Aura2 won't rely on `pacman.conf` so we can fix that problem from the ground up.

**Comment by flamusdiu** [comment]
@fosskers side question: how will the pacman.conf affect Aura2? Do users need to setup their repos again in whatever config file for Aura2 to work with them?

**Comment by fosskers** [comment]
KISS would tell us to have everything in one config file. Certainly Aura will have its own config. The raises the question of where `hook` configs should go. Perhaps leave that to the hooks completely? If so, we could just co-op `pacman.conf` for the Arch version of Aura2.

**Comment by flamusdiu** [comment]
Yes, KISS does say that but at the same time; people have a habit of setting up repos in pacman.conf. Maybe use them in both places.

**Comment by fosskers** [comment]
AUR packages don't have groups, am I right?

**Comment by mrak** [comment]
Correct, the AUR packages don't, but the same package that I installed manually using my own PKGBUILD _does_ have the group in question. It works for official packages that I have manually packaged myself, but `aura` will still install an AUR package over mine even when it was installed with the ignored group on it locally.

**Comment by fosskers** [comment]
Ah, I see now:
- Package `A` given group `G` in your custom build process
- Package `A` has an AUR version too
- `G` is marked as ignored, but since the "original" `A` has no group, it doesn't know to be ignored

Can you provide an example of such a package, and what the output of `-Qi` is for that? I'd like to see the group field.

**Comment by fosskers** [comment]
Detecting this might not be that complicated.

- Observe `IgnoreGroup` field. 
- Ask pacman which installed packages make up those groups
- Union these lists (via `Set`)
- For candidate installations / upgrades, check if they're part of that `Set`

**Comment by fosskers** [comment]
Also, thanks for getting back to me. I realize this issue has been open for years. I should haven been quicker about it.

**Comment by mrak** [comment]
Here is one example, my custom configuration of `ttf-input`:

```
$ fgrep IgnoreGroup /etc/pacman.conf
IgnoreGroup = modified

$ pacman -Qi ttf-input | fgrep Groups
Groups        : modified

**Comment by fosskers** [comment]
Alright, that should do it! These changes will be included in Aura 2.0, to be released soon (not today, mind you).
